pkg/asset/installconfig/azure/session.go

```diff
@@ -0,0 +1,155 @@
+package azure
```

Comment:
i want this file to include:

1. Credentials and getCredentials:
   `getCredentials` prompts user for bare minimum to setup the FileBasedAuthentication to deafult location.
2. Session and GetSession
    GetSession -> sets up the authorizer using file based or env based and also store the Credentials so that consumers can pass them along if needed.
    Session should provide Authorizer and Credentials as it looks like having Authorizer doesn't provide a way to get the Credentials that were used to setup the Authorizer so that we can pass them along into the cluster.

Reply:
I can reorganize this.
there is a public GetCredential function to read the creds as well, to pass them into the secret file template in fact.

Reply:
1. => why is it necessary for azure to prompt user for the file location? aws provider doesn't prompt

Reply:
I think this is making things overly complicated for a first version.
if we want to try all the supported source of credentials, rely on autorest, and provide the settings used to the cluster, the we need to roll our own. if you rely on built-in autorest functionality, to get the authorizer from env or from file, then you cannot reliably know which parameter where used. it's abstracted from you.

i.e. : Was it using the managed identity? Client Auth? Certificate? from env? or from file?

So to achieve what you ask, you need to use lower primitives, code the fallback logic yourself, plus retrieve the authorizer from these creds and store them where you want.
If you want to customize where the file is stored, then we need to save that file location for future uses too. It's all too much, and I don't think it is worth it to spend time on these details right now.

The current implementation is using a default file location, and asking for client creds if it's not there.
IMO, It is enough until we get to a working solution on azure

Reply:
Resolving as won't change in this iteration